### PR TITLE
input locations: check if specified path exists

### PR DESCRIPTION
**Pull request by swissiety**
+ add DefaultRTJarAnalysisInputLocation which uses the default Location of the rt.jar via java_home

### CODE REVIEW

File: sootup.core/src/main/java/sootup/core/signatures/MethodSubSignature.java [hunk 1]
```diff
@@ -134,4 +134,22 @@ public void toString(StmtPrinter printer) {
 
     printer.literal(")");
   }
+
+  boolean isMainSignature() {
```
Comment (kadirayk) [inline sootup.core/src/main/java/sootup/core/signatures/MethodSubSignature.java]: 1. Do you really want to have Java-specific methods in the core?
2. Should these be public?

File: sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/PathBasedAnalysisInputLocation.java [hunk 2]
```diff
@@ -789,4 +796,20 @@ public List<String> retrieveServletClasses(String extractedWARPath) {
       return classesInXML;
     }
   }
+
+  /**
+   * Refers to the rt.jar from <=Java8 as an AnalysisInputLocation requires: JAVA_HOME to be set and
+   * expects the jar in the "lib/" subdirectory. If you need to include the rt.jar from a custom
+   * Location please make use of JavaClassPathAnalysisInputLocation.
+   */
+  private static class DefaultRTJarAnalysisInputLocation extends ArchiveBasedAnalysisInputLocation {
```
Comment (kadirayk) [inline sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/PathBasedAnalysisInputLocation.java]: 1. shouldn't this be public? 
2. Also should we now switch to this everywhere we add rt.jar manually?

File: sootup.core/src/main/java/sootup/core/signatures/MethodSubSignature.java (hunk 1 above)
Comment (swissiety) [reply sootup.core/src/main/java/sootup/core/signatures/MethodSubSignature.java]: there was something.. other ~~unsupported~~ possible languages.. we can have another structure like JavaLanguage where  we can store this, (java.base/)`java.lang.throwable`, `java.lang.Object` etc.?

Comment (swissiety) [reply sootup.core/src/main/java/sootup/core/signatures/MethodSubSignature.java]: it affects Identifiers moved it to the JavaIdentifierFactory

File: sootup.java.core/src/main/java/sootup/java/core/JavaIdentifierFactory.java [hunk 3]
```diff
@@ -56,6 +56,32 @@ public class JavaIdentifierFactory implements IdentifierFactory {
 
   @Nonnull private static final JavaIdentifierFactory INSTANCE = new JavaIdentifierFactory();
 
+  public boolean isStaticInitializerSignature(@Nonnull MethodSubSignature methodSubSignature) {
+    return methodSubSignature.getName().equals("<clinit>");
```
Comment (JonasKlauke) [inline sootup.java.core/src/main/java/sootup/java/core/JavaIdentifierFactory.java:61]: there is a static String for this JavaSootMethod.STATIC_INITIALIZER_NAME

File: sootup.java.core/src/main/java/sootup/java/core/JavaIdentifierFactory.java [hunk 4]
```diff
@@ -56,6 +56,32 @@ public class JavaIdentifierFactory implements IdentifierFactory {
...
+  }
+
+  public boolean isStaticInitializerSignature(@Nonnull MethodSignature methodSignature) {
+    return isStaticInitializerSignature(methodSignature.getSubSignature());
+  }
+
+  public boolean isConstructor(@Nonnull MethodSubSignature methodSubSignature) {
+    return methodSubSignature.getName().equals("<init>");
```
Comment (JonasKlauke) [inline sootup.java.core/src/main/java/sootup/java/core/JavaIdentifierFactory.java:66]: there is a static String for this JavaSootMethod.CONSTRUCTOR_NAME

**Review by JonasKlauke — APPROVED** [review]
👍

**Comment by JonasKlauke** [comment]
put isConstructor, isStaticInitializer and isMainMethod in a Test to improve the Coverage 😄

**Review by kadirayk — APPROVED** [review]

File: sootup.java.core/src/main/java/sootup/java/core/JavaIdentifierFactory.java (hunk 3 above)
Comment (swissiety) [reply sootup.java.core/src/main/java/sootup/java/core/JavaIdentifierFactory.java:61]: moved it to the javaIDF

**Pull request merged**